### PR TITLE
[WIP] Synths V3.0 - Bob Shadowban Edition

### PR DESCRIPTION
**Pull request by CinderWC**
## About The Pull Request

Basically balances synths so they can be a roundstart race, in a way. Synths are kind of funky so i'll have to do some weird fucking shit to make this work.
Basically, what i did is make it a fucking quirk - This may seem weird, but it is way easier for me since the synth species inherits shit from another species on being gained.
This also helps in one way, as you'll be able to tell who is a synth with a health analyzer by the quirk - but not just by normally examining.
Synths do not take clone damage, because it would be impossible to heal.
Synthflesh heals burn, brute and toxin for Synths. But they do not process any other chems.
Synths cannot use ointment or bruise packs.
Synths, and only synths, are able to overdose on Synthflesh.
Synth limbs disable and dismember easier.

## Why It's Good For The Game

There **is** some demand for it, i see a lot of people roleplaying as synths without the proper tools. This will help with character building and include fun mechanics.

## Changelog
:cl:
add: Added a synth neutral quirk... which turns you into a synth once you spawn. Upsides and downsides are detailed in the desc. Fallout jokes yada yada yeah. See PR #1342 .
balance: Synths have been rebalanced for the synth quirk. Now they can be dismembered and their limbs can be disabled, and they breathe. They don't process chems or viruses, and synthflesh heals also heals toxin damage for them. Synths don't take clone damage.
refactor: Synth species datum and procs have been changed a bit so custom limb icons are taken into account and etc.
refactor: Scream emote now uses species vars and is generally more efficient.
fix: Fix for Radsuits not appearing
/:cl:

### CODE REVIEW

**Comment by CinderWC** [comment]
Correction: Not a typo fix, but makes synths process blood now

File: modular_skyrat/code/modules/mob/living/carbon/human/species_types/synths.dm [hunk 1]
```diff
@@ -28,6 +28,9 @@
 	mutanteyes = /obj/item/organ/eyes/ipc
 	exotic_blood = /datum/reagent/blood/synthetics
 	exotic_bloodtype = "SY"
+	//FUCKING STUPID PIECE OF SHIT STREET FUCKING CATS
```
Comment (CinderWC) [inline modular_skyrat/code/modules/mob/living/carbon/human/species_types/synths.dm:31]:
```suggestion
// cheeto
```

File: modular_skyrat/code/modules/mob/living/carbon/human/species_types/synths.dm [hunk 2]
```diff
@@ -51,6 +54,18 @@
 		hair_color = S.hair_color
 		screamsounds = S.screamsounds.Copy()
 		femalescreamsounds = S.femalescreamsounds.Copy()
+		if(istype(S, /datum/species/human/felinid)) //FUCK YOU, FUCKING CAT IDIOTS I FUCKING HATE  YOU I HOPE YOUR FUCKING
```
Comment (CinderWC) [inline modular_skyrat/code/modules/mob/living/carbon/human/species_types/synths.dm:57]:
```suggestion
		if(istype(S, /datum/species/human/felinid)) //bad cat
```

File: modular_skyrat/code/modules/mob/living/carbon/human/species_types/synths.dm [hunk 3]
```diff
@@ -51,6 +54,18 @@
...
+			storedeardamage = H.getOrganLoss(ORGAN_SLOT_EARS) //AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA TAIL GETS CUT OFF
```
Comment (CinderWC) [inline modular_skyrat/code/modules/mob/living/carbon/human/species_types/synths.dm:58]:
```suggestion
			storedeardamage = H.getOrganLoss(ORGAN_SLOT_EARS) // tail go bye
```

File: modular_skyrat/code/modules/mob/living/carbon/human/species_types/synths.dm [hunk 4]
```diff
@@ -51,6 +54,18 @@
...
+			mutantears = S.mutantears						//YOUR FUCKING EARS GET MUNCHED OFF BY A RABID DOG
```
Comment (CinderWC) [inline modular_skyrat/code/modules/mob/living/carbon/human/species_types/synths.dm:59]:
```suggestion
			mutantears = S.mutantears						//ears go bye
```

File: modular_skyrat/code/modules/mob/living/carbon/human/species_types/synths.dm [hunk 5]
```diff
@@ -131,6 +146,16 @@
 	meat = initial(meat)
 	limbs_id = initial(limbs_id)
 	use_skintones = initial(use_skintones)
+	if(istype(fake_species, /datum/species/human/felinid)) //worthless fFUCKING DINGY CATS
```
Comment (CinderWC) [inline modular_skyrat/code/modules/mob/living/carbon/human/species_types/synths.dm:149]:
```suggestion
	if(istype(fake_species, /datum/species/human/felinid)) //worthless catte
```

File: code/game/objects/items/stacks/medical.dm [hunk 6]
```diff
@@ -32,18 +32,20 @@
 
 /obj/item/stack/medical/proc/heal_carbon(mob/living/carbon/C, mob/user, brute, burn)
 	var/obj/item/bodypart/affecting = C.get_bodypart(check_zone(user.zone_selected))
+	var/datum/species/carbonspecies = C.dna.species
```
Comment (Trilbyspaceclone) [inline code/game/objects/items/stacks/medical.dm:35]:
```suggestion
	var/datum/species/carbonspecies = C.dna.species //Skyrat addtion
```

File: code/game/objects/items/stacks/medical.dm [hunk 7]
```diff
@@ -32,18 +32,20 @@
...
 	if(!affecting) //Missing limb?
 		to_chat(user, "<span class='warning'>[C] doesn't have \a [parse_zone(user.zone_selected)]!</span>")
 		return
 	if(affecting.status == BODYPART_ORGANIC) //Limb must be organic to be healed - RR
-		if(affecting.brute_dam && brute || affecting.burn_dam && burn)
```
Comment (Trilbyspaceclone) [inline code/game/objects/items/stacks/medical.dm:39]: all of this basicly needs citation of skyrat changes

File: code/modules/surgery/bodyparts/bodyparts.dm [hunk 8]
```diff
@@ -164,8 +164,8 @@
 		return FALSE
 
 	switch(animal_origin)
-		if(ALIEN_BODYPART,LARVA_BODYPART) //aliens take some additional burn //nothing can burn with so much snowflake code around
-			burn *= 1.2
+		if(ALIEN_BODYPART,LARVA_BODYPART) //aliens take double burn //nothing can burn with so much snowflake code around
```
Comment (Trilbyspaceclone) [inline code/modules/surgery/bodyparts/bodyparts.dm:167]:
```suggestion
		if(ALIEN_BODYPART,LARVA_BODYPART) //aliens take double burn //nothing can burn with so much snowflake code around //Skyrat changes, buffs from 1.2 to 2
```

File: modular_skyrat/code/modules/mob/living/carbon/human/species_types/synths.dm [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+	inherent_traits = initial_inherent_traits.Copy()
+	attack_verb = initial(attack_verb)
+	attack_sound = initial(attack_sound)
+	miss_sound = initial(miss_sound)
+	nojumpsuit = initial(nojumpsuit)
+	no_equip = list()
+	meat = initial(meat)
+	limbs_id = initial(limbs_id)
+	use_skintones = initial(use_skintones)
+	if(istype(fake_species, /datum/species/human/felinid)) //worthless catte
```
Comment (Trilbyspaceclone) [inline modular_skyrat/code/modules/mob/living/carbon/human/species_types/synths.dm:149]:
```suggestion
	if(istype(fake_species, /datum/species/human/felinid)) //Organs added by felinid race get added back on
```

File: modular_skyrat/code/modules/mob/living/carbon/human/species_types/synths.dm [hunk 10]
```diff
@@ -0,0 +1,195 @@
...
+		nojumpsuit = S.nojumpsuit
+		no_equip = S.no_equip.Copy()
+		icon_limbs = S.icon_limbs //there you go bubsy, now dunmer synths and shit wont be FUCKING INVISIBLE
+		limbs_id = S.limbs_id
+		use_skintones = S.use_skintones
+		fixed_mut_color = S.fixed_mut_color
+		hair_color = S.hair_color
+		screamsounds = S.screamsounds.Copy()
+		femalescreamsounds = S.femalescreamsounds.Copy()
+		if(istype(S, /datum/species/human/felinid)) //bad cat
```
Comment (Trilbyspaceclone) [inline modular_skyrat/code/modules/mob/living/carbon/human/species_types/synths.dm:57]:
```suggestion
		if(istype(S, /datum/species/human/felinid)) //Felinid looses their tails and ears when harmed to much
```

**Comment by CinderWC** [comment]
Thanks for all the good work Bob.

And thank you Useroth for merging it.

**Comment by krumpmeboyz** [comment]
Currently, synths can't replace their blood if they bleed out and then suffocate profusely for the rest of the shift if revived. Adding more synthblood doesn't do much if anything to solve the problem, as it seemingly just vanishes upon entering their bloodstream.

**Pull request merged**